src/Features/Core/Portable/Testing/TestFrameworks/MSTestTestFrameworkMetadata.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.Features.Testing;
+
+[Export(typeof(ITestFrameworkMetadata)), Shared]
+internal class MSTestTestFrameworkMetadata : ITestFrameworkMetadata
+{
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public MSTestTestFrameworkMetadata()
+    {
+    }
+
+    public bool MatchesAttributeSyntacticName(string attributeSyntacticName)
+    {
+        return attributeSyntacticName is "TestMethodAttribute";
```

Comment:
so, instead of having the caller add 'Attribute' this could just be:
```suggestion
        return attributeSyntacticName is "TestMethod" or "TestMethodAttribute";
```

which would be alloc free.

Reply:
you could also defer to some helper in a base type that does this check for you given the prefix.